### ISSUE TITLE
[ Install Issue ] Could not find C_MATH_LIBRARY using the following names: m

### ISSUE THREAD

**Issue by Afei027** [isce-framework/snaphu-py#60]
my os: win11
when i use pycharm to install , error like:
![1713333534896](https://github.com/isce-framework/snaphu-py/assets/92086866/d12290f0-cb4c-49b5-b751-e4de786255e8)
![1713333578243](https://github.com/isce-framework/snaphu-py/assets/92086866/9e305937-34e6-417d-9856-7f26e52c488a)

**Comment by scottstanie** [comment]
The [SNAPHU code](https://web.stanford.edu/group/radar/softwareandlinks/sw/snaphu/) from Stanford which this repository wraps does not compile on Windows. There have been some efforts to change the C code to make it work (e.g. https://github.com/PCIGeomatics/snaphu) but we haven't tried to incorporate that here.

**Comment by gmgunter** [comment]
Yeah, the official SNAPHU releases only support Unix-like platforms. The specific build error that you're getting is saying that it can't find the C math library `libm` (which is typically packaged with your C distribution on Unix systems), but SNAPHU also relies on some POSIX header files that aren't available on Windows (as far as I'm aware).

You might be able to get it working if you trying installing under the [WSL](https://learn.microsoft.com/en-us/windows/wsl/install). I'd be interested to hear how it goes if you give it a try.

**Comment by Afei027** [comment]
Thank you, I'll give it a try




Yの影子
***@***.***



&nbsp;




------------------ 原始邮件 ------------------
发件人:                                                                                                                        "isce-framework/snaphu-py"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年4月18日(星期四) 凌晨2:40
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [isce-framework/snaphu-py] [ Install Issue ] Could not find C_MATH_LIBRARY using the following names: m (Issue #60)





 
Yeah, the official SNAPHU releases only support Unix-like platforms. The specific build error that you're getting is saying that it can't find the C math library libm (which is typically packaged with your C distribution on Unix systems), but SNAPHU also relies on some POSIX header files that aren't available on Windows (as far as I'm aware).
 
You might be able to get it working if you trying installing under the WSL. I'd be interested to hear how it goes if you give it a try.

**Comment by Afei027** [comment]
Thank you, I'll take a look
&nbsp;


Yの影子
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Scott ***@***.***&gt;; 
发送时间: 2024年4月17日(星期三) 晚上8:19
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [isce-framework/snaphu-py] [ Install Issue ] Could not find C_MATH_LIBRARY using the following names: m (Issue #60)














 
The SNAPHU code from Stanford which this repository wraps does not compile on Windows. There have been some efforts to change the C code to make it work (e.g. https://github.com/PCIGeomatics/snaphu) but we haven't tried to incorporate that here.
